src/dotnet/install.sh

```diff
@@ -4,489 +4,133 @@
+if [ "$(id -u)" -ne 0 ]; then
+    echo -e 'Script must be run as root. Use sudo, su, or add "USER root" to your Dockerfile before running this script.'
     exit 1
 fi
 
-# Install the .NET CLI
-echo "(*) Installing .NET CLI..."
+# icu-devtools includes dependencies for .NET
+check_packages wget ca-certificates icu-devtools
 
-CHANGE_OWNERSHIP="false"
-if [[ "${DOTNET_ARCHIVE_ARCHITECTURES}" = *"${architecture}"* ]] && [[  "${DOTNET_ARCHIVE_VERSION_CODENAMES}" = *"${VERSION_CODENAME}"* ]] && [[ "${INSTALL_USING_APT}" = "true" ]]; then
-    echo "Detected ${VERSION_CODENAME} on ${architecture}. Attempting to install dotnet from apt"
+wget -O "$DOTNET_INSTALL_SCRIPT" "$DOTNET_INSTALL_SCRIPT_URL"
+chmod +x "$DOTNET_INSTALL_SCRIPT"
 
-    install_using_apt "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_VERSION}" false || install_using_apt "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_VERSION}" true
-    if [ "$?" != 0 ]; then
-        echo "Could not install requested version from apt on current distribution."
-        exit 1
-    fi
-else
-    if [[ "${INSTALL_USING_APT}" = "false" ]]; then
-        echo "Installing dotnet from releases url"
-    else
-        echo "Attempting to install dotnet from releases url"
-    fi
-    install_using_dotnet_releases_url "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_VERSION}"
-    CHANGE_OWNERSHIP="true"
-fi
+# Install primary version
+install_version "$DOTNET_VERSION"
 
-# Additional dotnet versions to be installed but not be set as default.
-if [ ! -z "${ADDITIONAL_VERSIONS}" ]; then
-    OLDIFS=$IFS
+# Install additional versions
+if [ -n "$DOTNET_ADDITIONAL_VERSIONS" ]; then
+    OLD_IFS=$IFS
     IFS=","
-        read -a additional_versions <<< "$ADDITIONAL_VERSIONS"
+        read -a additional_versions <<< "$DOTNET_ADDITIONAL_VERSIONS"
         for version in "${additional_versions[@]}"; do
-            OVERRIDE_DEFAULT_VERSION="false"
-            install_using_dotnet_releases_url "${DOTNET_SDK_OR_RUNTIME}" "${version}"
+            # Trim whitespace from version
+            version="$(echo -e "$version" | tr -d '[:space:]')"
+            install_version "$version"
         done
-    IFS=$OLDIFS
-fi
-
-if [ "${CHANGE_OWNERSHIP}" = "true" ]; then
-    if ! cat /etc/group | grep -e "^dotnet:" > /dev/null 2>&1; then
-        groupadd -r dotnet
-    fi
-    usermod -a -G dotnet "${USERNAME}"
-
-    chown -R "${USERNAME}:dotnet" "${TARGET_DOTNET_ROOT}"
-    chmod -R g+r+w "${TARGET_DOTNET_ROOT}"
-    find "${TARGET_DOTNET_ROOT}" -type d -print0 | xargs -n 1 -0 chmod g+s
-fi
-
-# Clean up
-if [ "${APT_PREFERENCES_UPDATED}" = "true" ]; then
-    if [ "${APT_PREFERENCES_FILE_ADDED}" = "true" ]; then
-        rm -f /etc/apt/preferences
-    else
-        head -n -3 /etc/apt/preferences > /tmp/preferences-temp && mv /tmp/preferences-temp /etc/apt/preferences
-    fi
+    IFS=$OLD_IFS
 fi
 
-rm -rf /var/lib/apt/lists/*
```

Comment:
Same here, it would be nice to clean up later. Can we keep this ?

Reply:
Yep, makes sense to me.